src/controllers/exams.controller.js

```diff
@@ -1,39 +1,73 @@
 
 const asyncHandler = require('express-async-handler');
 const Exam = require('../models/exams.model');
+const Question = require("../models/questions.model")
+const Response = require("../models/responses.model")
+const NotFoundError = require("../../Errors/NotFoundError");
 
 const getAllExams = asyncHandler(async (req, res, next) => {
-    const { populate, min ,max } = req.query;
-
-    res.status(200).send({
-        "success": true,
-        
-    });
+    try {
+        const { populate, min ,max } = req.query;
+        let query = {}
+        if (min) query.score = { $gte: min };
+        if (max) query.score = { $lte: max };
+    
+        const exams = await Exam.find(query).populate(populate); 
```

Comment:
let's keep it  simple for now     `const exams = await Exam.find({}).populate(populate); `